File: kanbanBoard-Back End/boards/src/main/java/com/niit/boards/model/Rule.java
package com.niit.boards.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Rule {
    @Id
    private int ruleId;
    //The column to move task from
    private int fromColumn;
    // The column to move task to
    private int toColumn;
    // The trigger field for task moving(can be task status, completion status, priority etc.)
    private String trigger;
    // The status of the trigger upon which the task will be moved
    private String triggerStatus;

}

File: kanbanBoard-Front End/src/app/classes/rule.ts
export class Rule {
    ruleId!:number
    // id of from column
    fromColumn!:number
    //id of to column
    toColumn!:number
    // the property used as trigger to move task(eg. priority/ completion status)
    trigger!:string
    //the status of the trigger upon which the rule will be applied
    triggerStatus!:string
}
